bridgetown-core/lib/bridgetown-core/frontmatter_defaults.rb

```diff
@@ -88,6 +95,16 @@ def all(path, type)
 
     private
 
+    def merge_data_cascade_for_path(path, merged_data)
+      absolute_path = @site.in_source_dir(path)
+      @site.defaults_reader.path_defaults
+        .select { |k, _v| absolute_path.include? k }
+        .sort_by { |k, _v| k.length }
+        .each do |defaults|
+          merged_data.merge!(defaults[1])
+        end
+    end
```

Comment:
So we basically first check that it exists, then we sort by length because in theory a cascade would look like this:

`src/_defaults.yml`, `src/_posts/_defaults.yml` so the file path will always be longer the further nested you go, pretty cool idea! and then you jsut merge in the defaults array. This is so slick and elegant.

Reply:
Right, it finds any matching parent paths (could be one, could be several…or none), and then does a progressive merge through the paths from highest level to lowest level. Getting those levels via string length sort is perhaps naive, but I can't think of a reason why that wouldn't work.

Reply:
I cant think of a reason either, if youre descending, it has to be longer.